src/shim.js

```diff
@@ -4,7 +4,89 @@
 
 "use strict";
 
-class UrlbarProvider {}
+class UrlbarProvider {
+  constructor(name) {
+    this.onBehaviorRequested = this.onBehaviorRequested.bind(this);
```

Comment:
Should we do a `let events = ["onBehaviorRequested", ....]` since there is a bunch of repitition here, also makes sure we remove everything we added etc